packages/plugins/response-cache/src/plugin.ts

```diff
@@ -13,11 +13,13 @@ import {
 import {
   ExecutionResult,
   getDocumentString,
+  IncrementalExecutionResult,
   isAsyncIterable,
   Maybe,
   ObjMap,
   Plugin,
 } from '@envelop/core';
+import { IncrementalDeferResult, IncrementalStreamResult } from '@graphql-tools/executor';
```

Comment:
Hmm if we really need these types, we can move them to `utils` package to avoid adding `executor` as a dependency.

Reply:
Oh yeah I didn't realized they were part of `executor`, I've added it for the tests, didn't meant to add it as a real dependency ><
So you think it is ok to just copy/paste it ?

Reply:
Do we really need that strict types for these two? Aren't they already inside `ExecutionResult` of `utils` package?

Reply:
Oh yes no I understood, you mean move it from `@graphql-tools/executor` to `@graphql-tools/utils`

Reply:
TS complains about `incremental` and `hasNext` not being declared on `ExecutionResult` ><. From the types, it seems `ExecutionResult` doesn't include `IncrementalResult`, which is true, you can't have an `IncrementalResult` as a direct result of execution.

Reply:
I just reviewed my code, and there is something I don't understand. In fact, we don't use `@graphql-tools/utils` types, but our own redefined types inside envelop. So I updated (but it seems I've failed to use it ^^') the envelop's types https://github.com/n1ru4l/envelop/pull/1896/files#diff-03f876d59dce271fae6bf2c434d7a111a2ebd06c787c73209dca96008bee8e23